src/main/java/com/infomaniak/lib/core/networking/NetworkAvailability.kt

```diff
@@ -32,24 +32,35 @@ import kotlinx.coroutines.channels.awaitClose
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.callbackFlow
 import kotlinx.coroutines.launch
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
 
 class NetworkAvailability(private val context: Context, private val ioDispatcher: CoroutineDispatcher = Dispatchers.IO) {
 
     private val connectivityManager by lazy { context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager }
+    private val mutex = Mutex()
 
     val isNetworkAvailable: Flow<Boolean> = callbackFlow {
         val networks = mutableListOf<Network>()
 
         val callback = object : NetworkCallback() {
 
             override fun onAvailable(network: Network) {
-                networks.add(network)
-                sendNetworkAvailability(networks)
+                launch {
```

Comment:
```suggestion
                launch(ioDispatcher) {
```

Reply:
It is on purpose that it runs on the default. 
As we're already in a flow, the default dispatcher will just be launched in background and do its thing.
The io has a special algorithm that wakes it after external read/write (explanation from sirambd), we don't want this here